### ISSUE TITLE
Feature Request: URL pre TSA server + casova peciatka

### ISSUE THREAD

**Issue by MarkoD** [slovensko-digital/autogram#323]
Ahojte, k dokolanej kompletnosti chyba este podpora pre casove peciatky + do nastaveni dat URL pre custom TSA server:
1. na SK su platene TSA servre,
2. v ramci EU je niekolko free TSA serverov, ktore (v ramci FUP) umoznuju kvalifikovanu casovu peciatku zadarmo

Podpora pre casove peciatky spolu s moznostou 2 by robili s Autogramu este dokonalejsi nastroj ;-)

**Comment by MarkoD** [comment]
Pripajam link na trusted qualified timestamp authorities v celej EU:

https://eidas.ec.europa.eu/efda/tl-browser/#/screen/search/type/3?searchCriteria=eyJjb3VudHJpZXMiOlsiQVQiLCJCRSIsIkJHIiwiQ1kiLCJDWiIsIkRFIiwiREsiLCJFRSIsIkVMIiwiRVMiLCJGSSIsIkZSIiwiSFIiLCJIVSIsIklFIiwiSVMiLCJJVCIsIkxJIiwiTFQiLCJMVSIsIkxWIiwiTVQiLCJOTCIsIk5PIiwiUEwiLCJQVCIsIlJPIiwiU0UiLCJTSSIsIlNLIiwiVUsiXSwicVNlcnZpY2VUeXBlcyI6WyJRVGltZXN0YW1wIl19

**Comment by jsuchal** [comment]
@MarkoD ano toto sme zamerne odkladali, lebo v SK realiach casove peciatky az tak velmi netreba. Mimimalne v komunikacii so statom si tu CP musi pridat urad sam, cize to nemusis riesit. Zmysel to (asi) dava pri obcianskom a obchodnom vztahu a tie FUP free TSA davaju velky zmysel. 

Tu vsak musime doriesit UX, zatial je moja predstava taka, ze budes mat okienko "ako to chces podpisat" - asi tam kde je teraz vyber certifikatu a tam si okrem klasickeho podpisu budes vediet vybrat aj taky s casovou peciatkou (=osvedceny podpis). Treba vsak domysliet dalsie veci. (napriklad v tom momente este nevieme ci dany certifikat je ku QES).

**Comment by MarkoD** [comment]
@jsuchal Dakujem za info. Poznam viacero firiem, ktori pouzivaju bud bezplatne TSA servre alebo si platia slovenske. Ked bude mozne toto nastavit v Autograme, mozu plnohodnotne prejst na Autogram ;-)

**Comment by jsuchal** [comment]
@MarkoD mozno to dame na dva kroky, prve bude take to klasicke nastavenie v settings co bude pre power users a potom ked domyslime ux a detaily tak to vytiahneme aj inde. Deal?

**Comment by MarkoD** [comment]
@jsuchal Jasne, kludne. Aj tak je toto nastavenie skor pre power users, ako pises ;-)

**Comment by celuchmarek** [comment]
@MarkoD Vo verzii v2.1.2 https://github.com/slovensko-digital/autogram/releases/tag/v2.1.2 sa už dá v nastaveniach určiť url TSA servera a zapnúť pridávanie časových pečiatok.

**Comment by MarkoD** [comment]
@celuchmarek Dakujem, vyskusam.

**Comment by anthoniussk** [comment]
Časové pečiatky vo verzii 2.1.2 sú fajn, od Ditecu ale máme časové pečiatky, ktoré vyžadujú pre prístup zadať meno a heslo. Dá sa to nastaviť niekde v Autograme?

**Comment by jsuchal** [comment]
@anthoniussk existuje nejaky dovod preco preferujes platene casove peciatky pred neplatenymi? Kazdy tento provider ma iny sposob autentifikacie cize je to dost narocne na implementaciu.

**Comment by anthoniussk** [comment]
@jsuchal u klienta to používajú, nepozeral som, či právnici môžu používať hocijaké. Dostávajú to zo SAKu od ica.cz

**Comment by vypinac** [comment]
Dobry den, dakujem za tento program.
Podpisal som PDF/A aj s casovou peciatkou a ked som ho dal overit na qesportal.sk, tak pise:

Stav podpisu
Podpis je platný (úplné overenie)   
Druh podpisu
Kvalifikovaný elektronický podpis **bez časovej pečiatky**
Overené k času
26. január 2024 02:15:11 (UTC)  

Časová pečiatka
Výsledok overenia
Nerozhodnutý
Čas
26. január 2024 02:14:17 (UTC)
Integrita
Nenarušená
Druh časovej pečiatky
Časová pečiatka podpisu
Podpísal - tsa.izenpe.com
Výsledok overenia
Integrita
Nenarušená
Hash algoritmus
SHA256
Šifrovací algoritmus
RSA
Podpisová politika
DSId
MDEwDQYJYIZIAWUDBAIBBQAEIOjHT/Brdv8bGHFTctmeYJ8busEUd5MbRNv8LSrGThLE
Dôvody výsledku overenia
Podpisový certifikát nebol overený ako plne platný.
Certifikačná cesta
[Certifikát - tsa.izenpe.com]
[Certifikát - SUBCA QC IZENPE - TSA]
[Certifikát - ROOT CA QC IZENPE]
Dôveryhodný zoznam
Identifikátor
336
URL
https://ec.europa.eu/tools/lotl/eu-lotl.xml
Teritórium
EU - Európska únia
Čas overenia
26. január 2024 02:15:01 (UTC)

Je niekde nieco zle? Dakujem

**Comment by jsuchal** [comment]
@vypinac ahoj, pravdepodobne si len nezapol casove peciatky / osvedceny podpis v nastaveniach (pozri paticku po zapnuti autogramu).  Potom by to malo normalne ist. Pripadne si vyber zdroj casovych peciatok.

**Comment by vypinac** [comment]
Zdravim, casove peciatky mam zapnute a peciatka tam je
zdroj peciatky napokon ukazuje aj overenie cez qesportal.sk:

Časová pečiatka
Výsledok overenia Nerozhodnutý
Čas 30. január 2024 02:58:42 (UTC)
Integrita Nenarušená
Druh časovej pečiatky Časová pečiatka podpisu
Podpísal - tsa.izenpe.com
Dôvody výsledku overenia Podpisový certifikát nebol overený ako plne platný.

Druh podpisu Kvalifikovaný elektronický podpis bez časovej pečiatky

bud autogram zle podpisuje (pridava peciatku) alebo qesportal zle overuje (nedokaze ju plne overit/rozoznat) ... 
mozete si to vyskusat, podpiste hociaky dokument cez autogram a skuste si ho overit cez qesportal...

**Comment by jsuchal** [comment]
Skusil som, ukazuje mi to to iste. Je to zvlastne. Lebo napriklad https://ec.europa.eu/digital-building-blocks/DSS/webapp-demo/validation neprotestuje, dokonca ani https://www.slovensko.sk/sk/e-sluzby/sluzba-overenia-zep neprotestuje. 

Problem na qes portali vyzera byt tato cast. 

![image](https://github.com/slovensko-digital/autogram/assets/26342/6785593c-e687-412a-bfcb-ce11f9bd7680)

**Comment by jsuchal** [comment]
@vypinac skusal si iny zdroj peciatok?

**Comment by celuchmarek** [comment]
Písal som ohľadom tohto qesportalu.

**Comment by celuchmarek** [comment]
Qesportal na toto v čase riešenia reagoval tým, že nemali aktualizované TL Španielska. Krátko po tomto vyriešení s ich supportom sa Izenpe pečiatky overovali ako platné. Dnes je situácia opäť taká, že sú na qesportali neplatné. Skúsime ich zase pingnúť, ale teda táto issue sa netýka priamo Autogramu 🙂

**Comment by durkovic** [comment]
Qesportal z principu neoveruje napr. ani zahranicne casove peciatky, ktore maju doveryhodnu sluzbu naviazanu nie priamo na TSU, ale nepriamo na TSA, ktora vydava certifikaty pre viacero TSU. Diskusia s Disig-om na tuto temu neviedla nikam, hoci mame stanovisko Europskej komisie ze taketo casove peciatky su platne.

<img width="1045" alt="Image" src="https://github.com/user-attachments/assets/ce83741c-b75f-483a-9c81-9c2c32fda332" />

**Comment by jsuchal** [comment]
@durkovic nevedel by si to stanovisko vyzdielat?

**Comment by durkovic** [comment]
Poslane ako PM

**Comment by vypinac** [comment]
Dobry den, Autogram 2.5.0, zapnute casove peciatky s default volbou, ked som po podpisani znova otvoril subor v Autograme, overilo podpisy a napisalo zelenym QESig a **zltym Neznama CP.**
Je to chyba daneho poskytovatela TSP?
Ovplyvnuje to platnost alebo podpis akoby overeny dodatocne notarom?
Vdaka


Timestamps 1
Timestamp T-15821C8642319415A34E896933A617623AF14C2575547AB7A8A2ECDDD2D64B86
Qualification:
TSA
Qualification Details:
**The certificate is not related to a TSA/QTST!**
Indication:
PASSED
Certificate Chain:
BalTstamp QTSA TSU1
GLOBALTRUST 2015 ADVANCED SEAL 1[e-commerce monitoring GmbH](https://eidas.ec.europa.eu/efda/tl-browser/#/screen/tl/trustmark/AT/VATAT-U54992708)
GLOBALTRUST 2015
Production time:
2025-07-17 19:06:14 (UTC)

**Comment by celuchmarek** [comment]
Jo, trik je v tom, že po zmene nastavení množiny trusted listov je potrebné reštartovať Autogram, aby sa toto nastavenie prejavilo.

Baltstamp je Litovský zdroj a teda táto krajina nebola default zapnutá v predošlých verziách Autogramu, takže je ju často potrebné zapnúť manuálne.
